baseplate/diagnostics/tracing/publisher.py

```diff
@@ -0,0 +1,191 @@
+        headers = {
+            "User-Agent": "baseplate-trace-publisher/1.0",
+            "Content-Type": "application/json",
+        }
+        tries = 0
+        while True:
+            try:
+                with self.metrics.timer("post"):
+                    response = self.session.post(
+                        self.endpoint,
+                        data=payload.bytes,
+                        headers=headers,
+                        timeout=self.post_timeout,
+                        stream=False
+                    )
+                response.raise_for_status()
+            except requests.HTTPError as exc:
+                self.metrics.counter("error.http").increment()
+                response = getattr(exc, "response", None)
+                if response is not None:
+                    logger.exception("HTTP Request failed. Error: %s", response.text)
+                    # If client error, crash
+                    if response.status_code < 500:
+                        raise
+                else:
+                    logger.exception("HTTP Request failed. Response not available")
+            except IOError:
+                self.metrics.counter("error.io").increment()
+                logger.exception("HTTP Request failed")
+            else:
+                self.metrics.counter("sent").increment(payload.count)
+                return
+
+            tries += 1
+            if tries > self.retry_limit:
+                logger.error("Retry limit hit")
+                raise RetryLimitExceeded()
+            else:
+                logger.info("Retrying batch send. Try number %s", tries)
+                time.sleep(self.retry_delay * tries)
+
+
+def publish_traces():
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument("config_file", type=argparse.FileType("r"),
+        help="path to a configuration file")
+    arg_parser.add_argument("--queue-name", default="main",
+        help="name of trace queue / publisher config (default: main)")
+    arg_parser.add_argument("--debug", default=False, action="store_true",
+        help="enable debug logging")
+    arg_parser.add_argument("--app-name", default="main", metavar="NAME",
+        help="name of app to load from config_file (default: main)")
+    args = arg_parser.parse_args()
+
+    if args.debug:
+        level = logging.DEBUG
+    else:
+        level = logging.WARNING
+    logging.basicConfig(level=level)
+
+    config_parser = configparser.RawConfigParser()
+    config_parser.readfp(args.config_file)
+
+    app_raw_config = dict(config_parser.items("app:{}".format(args.app_name)))
+    app_cfg = config.parse_config(app_raw_config, {
+        "tracing": {
+            "endpoint": config.Optional(config.Endpoint),
+            "queue_name": config.Optional(config.String)
+        }
+    })
+
+    # pylint: disable=maybe-no-member
+    assert args.queue_name == app_cfg.tracing.queue_name, \
```

Comment:
Why have this on the command line and the config file if they must match?

Reply:
bump

Reply:
I thought I responded to this...

I left the command line arg for queue name to make the interface consistent with the event publisher. I'll remove the command line arg, I suppose.

Reply:
Or remove it from the config to match the event publishers?

Reply:
Oh... I was using the [listing service config](https://github.com/reddit/puppet/blob/master/modules/reddit_listing_server/templates/listing.ini.erb#L29) as a reference. It has an ``events.queue_name`` but now taking a closer look at the code, it doesn't seem to be used anywhere.